### PR TITLE
Rewrite keycode handling (keycode.cpp)

### PR DESCRIPTION
**Pull request by sfan5**
Changes:
- It no longer hurts to look at the code
- **Unittests**
- You can now bind to keys like `#`

Checklist:
- [x] Verify it still works on Linux
- [x] Verify it still works on Windows

### CODE REVIEW

File: src/keycode.cpp [hunk 1]
```diff
@@ -31,321 +32,329 @@ class UnknownKeycode : public BaseException
...
+struct table_key {
+	const char *Name;
+	irr::EKEY_CODE Key;
+	wchar_t Char; // may be L'\0' which means no char assigned
+	const char *LangName; // NULL means it doesn't have a human description
+};
+
+#define DEFINEKEY1(x, lang) { #x, irr::x, L'\0', lang }, // Irrlicht key without char
+#define DEFINEKEY2(x, ch, lang) { #x, irr::x, ch, lang }, // Irrlicht key with char
+#define DEFINEKEY3(ch) { "KEY_KEY_" TOSTRING(ch), irr::KEY_KEY_ ## ch, (wchar_t) *TOSTRING(ch), TOSTRING(ch) }, // single Irrlicht key (e.g. KEY_KEY_X)
```
Comment (est31) [inline src/keycode.cpp]: Can you make these macros multiline? They are over the 90 chars limit

**Comment by est31** [comment]
Concept wise I like this change, the old code indeed was to look bad at.

**Comment by Zeno-** [comment]
:+1:

**Comment by sfan5** [comment]
c3e1ab859e310f60c0462428d3c991b5218c98d2
cc0b3c1cd14f036d6336ae0fb2329fd245b6b4b4

**Pull request closed without merging**